src/factory.js

```diff
@@ -92,11 +96,16 @@ export default function factory(React, empty) {
           const wasAuthorized = isAuthorized(this.props.authData)
           const wasAuthenticating = this.props.isAuthenticating
 
-          if ( // Redirect if:
-              // 1. Was authorized, but no longer and not currently authenticating
-              (wasAuthorized && !willBeAuthorized && !willbeAuthenticating) ||
-              // 2. Was not authorized and authenticating but no longer authenticating
-              (wasAuthenticating && !willbeAuthenticating && !willBeAuthorized)
+          // Don't bather to redirect if:
+          // 1. currently authenticating or FailureComponent is set
+          if (willbeAuthenticating || !isRedirect)
+            return
+
+          // Redirect if:
+          if ( // 1. Was authorized, but no longer
+              (wasAuthorized && !willBeAuthorized) ||
+              // 2. Was not authorized and authenticating
+              (wasAuthenticating && !willBeAuthorized)
             ) {
```

Comment:
Is this correct? Seems like we need an extra parenthesis wrapping the `||` expression.